### PR TITLE
automatic renewal and recovery

### PR DESCRIPTION
**Pull request by tiero**
It closes https://github.com/arkade-os/ts-sdk/issues/19

## Summary by CodeRabbit

- New Features
  - Added VTXO Renewal and Recovery utilities to manage expiring and recoverable balances, with optional callbacks.
  - Wallet now supports renewal configuration and exposes getExpiringVtxos and renewVtxos.
  - Recovery and Renewal are available in the public API.

- Documentation
  - Expanded guides for VTXO Renewal and Recovery with usage examples and callbacks.
  - Simplified Transaction History examples.

- Bug Fixes
  - More consistent VTXO lists by unifying filtering logic (handles subdust and recoverable flags correctly).

- Tests
  - Added comprehensive unit tests for Renewal and Recovery workflows and edge cases.

### CODE REVIEW

File: src/wallet/serviceWorker/worker.ts [hunk 1]
```diff
@@ -470,22 +470,28 @@ export class Worker {
...
-                    return dustAmount == null
-                        ? true
-                        : !isSubdust(v, dustAmount);
-                };
-                vtxos = vtxos
-                    .filter(notSubdust)
-                    .filter((v) => !isRecoverable(v));
-            }
+             const vtxos = await this.getSpendableVtxos();
+
```
Comment (Kukks) [inline src/wallet/serviceWorker/worker.ts:474]:
```suggestion
```

File: src/wallet/recovery.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+/**
+ * Get recoverable VTXOs including subdust coins if their combined value exceeds dust threshold
+ *
+ * @param vtxos - Array of virtual coins to check
+ * @param dustAmount - Dust threshold amount in satoshis
+ * @returns Object containing recoverable VTXOs and whether subdust should be included
+ */
+function getRecoverableWithSubdust(
```
Comment (Kukks) [inline src/wallet/recovery.ts:23]: This doesn't make sense: it's the total that matters, not the subdust total.

File: src/wallet/recovery.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+import { SettlementEvent } from "../providers/ark";
+
+/**
+ * Filter VTXOs that are recoverable (swept and still spendable)
+ *
+ * @param vtxos - Array of virtual coins to check
+ * @returns Array of recoverable VTXOs
+ */
+function getRecoverableVtxos(
+    vtxos: ExtendedVirtualCoin[]
```
Comment (Kukks) [inline src/wallet/recovery.ts:11]: do Ark Notes get included in this @louisinger?

File: src/wallet/wallet.ts [hunk 4]
```diff
@@ -960,6 +975,87 @@ export class Wallet implements IWallet {
...
+        eventCallback?: (event: SettlementEvent) => void
+    ): Promise<string> {
+        // Get all VTXOs (including recoverable ones)
+        const vtxos = await this.getVtxos({ withRecoverable: true });
+
+        if (vtxos.length === 0) {
+            throw new Error("No VTXOs available to renew");
+        }
+
+        const totalAmount = vtxos.reduce((sum, vtxo) => sum + vtxo.value, 0);
```
Comment (Kukks) [inline src/wallet/wallet.ts:1042]: check  if it is above dust with totalAmount.

**Review by Kukks — CHANGES_REQUESTED** [review]
I would probably consolidate into a single class, which  defines behavior for recovering and renewing in one go.
